app/controllers/settings_controller.rb

```diff
@@ -4,63 +4,60 @@ class SettingsController < ApplicationController
     controller.ensure_logged_in t("layouts.notifications.you_must_log_in_to_view_your_settings")
   end
 
-  before_filter :except => :unsubscribe do |controller|
-    controller.ensure_authorized t("layouts.notifications.you_are_not_authorized_to_view_this_content")
-  end
+  before_filter EnsureCanAccessPerson.new(:person_id, error_message_key: "layouts.notifications.you_are_not_authorized_to_view_this_content"), except: :unsubscribe
 
   def show
-    flash.now[:notice] = t("settings.profile.image_is_processing") if @current_user.image.processing?
+    target_user = Person.find_by_username_and_community_id(params[:person_id], @current_community.id)
+    add_location_to_person!(target_user)
+    flash.now[:notice] = t("settings.profile.image_is_processing") if target_user.image.processing?
     @selected_left_navi_link = "profile"
-    add_location_to_person
+    render locals: {target_user: target_user}
   end
 
   def account
+    target_user = Person.find_by_username_and_community_id(params[:person_id], @current_community.id)
     @selected_left_navi_link = "account"
-    @person.emails.build
-    marketplaces = @person.community_memberships
+    target_user.emails.build
+    marketplaces = target_user.community_memberships
                    .map { |m| Maybe(m.community).name(I18n.locale).or_else(nil) }
                    .compact
-    has_unfinished = TransactionService::Transaction.has_unfinished_transactions(@current_user.id)
+    has_unfinished = TransactionService::Transaction.has_unfinished_transactions(target_user.id)
 
-    render locals: {marketplaces: marketplaces, has_unfinished: has_unfinished}
+    render locals: {marketplaces: marketplaces, has_unfinished: has_unfinished, target_user: target_user}
   end
 
   def notifications
+    target_user = Person.find_by_username_and_community_id(params[:person_id], @current_community.id)
     @selected_left_navi_link = "notifications"
-  end
-
-  def payments
-    @selected_left_navi_link = "payments"
+    render locals: {target_user: target_user}
   end
 
   def unsubscribe
```

Comment:
I noticed this method is omitted from the actions that the EnsureCanAccessPerson filter applies to and started to look where is it checked who has authority to do this operation on behalf of a person but can't find any check. Is there any?